services/route_manager/src/main/java/com/futurewei/alcor/route/service/Impl/NeutronRouterServiceImpl.java

```diff
@@ -0,0 +1,484 @@
+        router.setRouteTable(routeTable);
+        this.routerDatabaseService.addRouter(router);
+
+        // Construct response
+        for (RouteEntry routeEntry : routeEntities) {
+            String destination = routeEntry.getDestination();
+            String nexthop = routeEntry.getNexthop();
+            RoutesToNeutronRouteObject routesToNeutronRouteObject = new RoutesToNeutronRouteObject(destination, nexthop);
+            responseRoutes.add(routesToNeutronRouteObject);
+        }
+        responseRouter.setId(routerid);
+        responseRouter.setName(router.getName());
+        responseRouter.setRoutes(responseRoutes);
+
+        return new RoutesToNeutronWebResponse(responseRouter);
+    }
+
+    @Override
+    public ConnectedSubnetsWebResponse getConnectedSubnets(String projectId, String vpcId, String subnetId) throws ResourceNotFoundException, ResourcePersistenceException, SubnetNotBindUniquePortId {
+        List<String> subnetIds = new ArrayList<>();
+
+        // get subnet
+        SubnetWebJson subnetWebJson = this.routerToSubnetService.getSubnet(projectId, subnetId);
+        SubnetEntity subnet = subnetWebJson.getSubnet();
+        if (subnet == null) {
+            logger.warn("can not find subnet by subnet id :" + subnetId);
+            return null;
+        }
+
+        // get subnet's route table type
+        String attachedRouterId = subnet.getAttachedRouterId();
+        if (attachedRouterId == null) {
+            return null;
+        }
+
+        Router router = this.routerDatabaseService.getByRouterId(attachedRouterId);
+        if (router == null) {
+            return null;
+        }
+        RouteTable routeTable = router.getRouteTable();
+        if (routeTable == null) {
+            return null;
+        }
+        RouteTableType routeTableType = routeTable.getRouteTableType();
+
+        if (routeTableType == null) {
+            return null;
+        } else if(routeTableType.getRouteTableType().equals("neutron")){
+            List<String> ports = router.getPorts();
+            // check ports
+            if (ports == null || ports.size() <= 1) {
+                return null;
+            }
+            for (String portId : ports) {
+                // get subnet by port id
+                SubnetsWebJson subnetsWebJson = this.routerToSubnetService.getSubnetsByPortId(projectId, portId);
+                if (subnetsWebJson == null) {
+                    return null;
+                }
+                ArrayList<SubnetEntity> subnets = subnetsWebJson.getSubnets();
+                if (subnets.size() == 0) {
+                    return null;
+                }
+                if (subnets.size() != 1) {
+                    throw new SubnetNotBindUniquePortId();
+                }
+                subnet = subnets.get(0);
+                subnetIds.add(subnet.getId());
+            }
+
+        } else {
+            // TODO: vpc route operation
+            return null;
```

Comment:
@kevin-zhonghao For VPC, it's simpler than Neutron. We just need to do the following tasks:
1) get router by VPC:id
2) get routetables from the router
3) get subnet-ids from each routetable
4) return all subnet-ids and router info